### ISSUE TITLE
Fixed LTI

### ISSUE THREAD

**Issue by Josnic** [chamilo/chamilo-lms#2592]
Hi!
I´m working with Chamilo 1.11.6 and I tried to use the LTI Plugin. It did not work. However, I found a solution based on the repository https://github.com/csev/sakai-lti-test. It contains a complete and functional OAuth library, an LTI utility that I have edited to remove the unused code and leave only the necessary one. I modified the form.php file to integrate everything and it worked for me. I want to contribute my solution to the community so that Chamilo integrates with LTI without problems. ZIP attachment with the necessary files and modifications of form.php. Just copy and paste inside the plugin/ims_lti directory.
[UpdateLTI.zip](https://github.com/chamilo/chamilo-lms/files/2201698/UpdateLTI.zip)

**Comment by meuhland** [comment]
Looks interesting afaik the main issues with LTI as it was back in 1.11.6 was that it was just a whole bunch of stuff appended to the querystring. Which you seem to be adressing also in your code

I've submitted a quick and dirty fix to switch it to POST keeping most of the original code intact.
#2504

And internally we are working to further the LTI integration to make it work as a learnPath item.

**Comment by Josnic** [comment]
Hi!
I implemented his code #2504 , but it did not work. That's why I decided to use the other OAuth library and change the form.php code.

**Comment by AngelFQC** [comment]
Hi @Josnic 

Your changes on plugin works fine, but the plugin currently is working good with the changes from #2504. Can you test the plugin in https://11.chamilo.org (admin/admin)?

Add a tool in a course and send us the url or screenshot. Like https://11.chamilo.org/plugin/ims_lti/start.php?id=10&cidReq=LATABLEDEMULTIPLICATIONX3&id_session=0&gidReq=0&gradebook=0&origin=

**Comment by Josnic** [comment]
Hi @AngelFQC 

Your external tool is working, but my tool don´t work. My url is https://11.chamilo.org/plugin/ims_lti/start.php?id=12&cidReq=PRUEBANIC&id_session=0&gidReq=0&gradebook=0&origin= 
See https://11.chamilo.org/plugin/ims_lti/admin.php 
![image](https://user-images.githubusercontent.com/13033658/43740999-806e8a20-9992-11e8-8962-ba2eeffe53d1.png). These are my data connection LTI.

Thanks you.

**Comment by AngelFQC** [comment]
Hi @Josnic 
Maybe the wrong is with your tool? Or does it need a special configuration or requirement? Please try with other LTI tool, with the current code on chamilo 1.11.x

**Comment by Josnic** [comment]
Hi @AngelFQC 

I tried an online tool because I thought the same as you: http://ltiapps.net/test/tc.php. However, I saw that it worked correctly and that there was something in the code. I did tests with the Moodle LTI and it also worked well. That is why I made the changes and I decided to share them with the community, thinking that perhaps the same thing happened to many users as it did to me.


My time is limited, but I want this to help the community and in turn give your team for to review the issue. When I have a little more time, I will help.

Thanks you.

**Comment by renowijoyo** [comment]
what is the the current working state of v1.5? I'm using it with taotesting tool provider and unable to have it working right out of the box. i checked the tao log and found 'cannot identify resource_link_id and add them in the advance setting box but still not working

**Comment by jacovm** [comment]
We discussed an issue with the LTI plugin a while ago in this thread. https://github.com/chamilo/chamilo-lms/issues/2378#issuecomment-369905413

In case this is still relevant, we tested our LTI tool with LTI v1.5 and it seems to be working fine. 

Thanks very much for the help on this.

Jacov
![Untitled](https://user-images.githubusercontent.com/24301344/60964108-26afa300-a31b-11e9-93e8-aa2ab9f67c53.png)
